src/components/Grid/Grid.tsx

```diff
@@ -212,6 +213,28 @@ export const Grid = forwardRef<HTMLDivElement, GridProps>(
       onCopyCallback,
     ]);
 
+    const rowHeightsRef = useRef(new Map());
+
+    const getRowHeight = useCallback(
+      (index: number) => {
+        if (rowAutoHeight && rowHeightsRef.current.get(index)) {
+          return rowHeightsRef.current.get(index) + 33;
+        }
+        return rowHeight;
+      },
+      [rowHeight, rowAutoHeight]
+    );
+
+    const updateRowHeight = useCallback((rowIndex: number, height: number) => {
+      const prevHeight = rowHeightsRef.current.get(rowIndex) || 0;
+      if (height > prevHeight) {
+        rowHeightsRef.current.set(rowIndex, height);
+        if (gridRef.current) {
+          gridRef.current.resetAfterRowIndex(rowIndex);
+        }
+      }
+    }, []);
+
```

Comment:
I think you should update `Grid.test.tsx` to handle the new cases introduced in this PR